src/D2L.CodeStyle.Annotations/Contract/OnlyVisibleToAttribute.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+
+namespace D2L.CodeStyle.Annotations.Contract {
+
+	[AttributeUsage(
+		validOn: AttributeTargets.Method | AttributeTargets.Property,

```

Comment:
Starting small.  Class & struct would make sense also, but don't have the immediate use case.